crates/matrix-sdk-ui/tests/integration/room_list_service.rs

```diff
@@ -111,94 +111,106 @@ macro_rules! entries {
                     }
                 );
             ]
         )
     };
 
     // `remove [$nth]`
-    ( @_ [ $stream:ident ] [ remove [ $index:literal ] ; $( $rest:tt )* ] [ $( $accumulator:tt )* ] ) => {
+    ( @_ [ $entries:ident ] [ remove [ $index:literal ] ; $( $rest:tt )* ] [ $( $accumulator:tt )* ] ) => {
         assert_entries_stream!(
             @_
-            [ $stream ]
+            [ $entries ]
             [ $( $rest )* ]
             [
                 $( $accumulator )*
                 assert_eq!(
-                    $stream.next().now_or_never(),
-                    Some(Some(VectorDiff::Remove { index: $index })),
+                    $entries.next(),
+                    Some(&VectorDiff::Remove { index: $index }),
                 );
             ]
         )
     };
 
     // `insert [$nth] [ $entry ]`
-    ( @_ [ $stream:ident ] [ insert [ $index:literal ] [ $( $entry:tt )+ ] ; $( $rest:tt )* ] [ $( $accumulator:tt )* ] ) => {
+    ( @_ [ $entries:ident ] [ insert [ $index:literal ] [ $( $entry:tt )+ ] ; $( $rest:tt )* ] [ $( $accumulator:tt )* ] ) => {
         assert_entries_stream!(
             @_
-            [ $stream ]
+            [ $entries ]
             [ $( $rest )* ]
             [
                 $( $accumulator )*
                 assert_matches!(
-                    $stream.next().now_or_never(),
-                    Some(Some(VectorDiff::Insert { index: $index, value })) => {
-                        assert_eq!(value, entries!( $( $entry )+ )[0]);
+                    $entries.next(),
+                    Some(&VectorDiff::Insert { index: $index, ref value }) => {
+                        assert_eq!(value, &entries!( $( $entry )+ )[0]);
                     }
                 );
             ]
         )
     };
 
     // `pending`
-    ( @_ [ $stream:ident ] [ pending ; $( $rest:tt )* ] [ $( $accumulator:tt )* ] ) => {
+    ( @_ [ $entries:ident ] [ pending ; $( $rest:tt )* ] [ $( $accumulator:tt )* ] ) => {
         assert_entries_stream!(
             @_
-            [ $stream ]
+            [ $entries ]
             [ $( $rest )* ]
             [
                 $( $accumulator )*
-                assert_eq!($stream.next().now_or_never(), None);
+                assert_eq!($entries.next(), None);
+                // assert_eq!($stream.next().now_or_never(), None);
             ]
         )
     };
 
-    ( @_ [ $stream:ident ] [] [ $( $accumulator:tt )* ] ) => {
+    ( @_ [ $entries:ident ] [] [ $( $accumulator:tt )* ] ) => {
         $( $accumulator )*
     };
 
     ( [ $stream:ident ] $( $all:tt )* ) => {
-        assert_entries_stream!( @_ [ $stream ] [ $( $all )* ] [] )
+        // Wait on Tokio to run all the tasks. It won't happen in the main app.
+        yield_now().await;
+
+        let entries = $stream
+            .next()
+            .now_or_never()
+            .expect("failed to read from the stream")
+            .expect("failed to read from the stream (bis)");
```

Comment:
Maybe we could use more descriptive text here:

- for the first one: `stream entry wasn't in the ready state`
- for the second one: `stream was stopped`